rust/kernel/miscdev.rs

```diff
@@ -14,41 +14,46 @@ use core::marker::PhantomPinned;
 use core::pin::Pin;
 
 /// A registration of a miscellaneous device.
-pub struct Registration {
+pub struct Registration<T: Sync = ()> {
     registered: bool,
     mdev: bindings::miscdevice,
     _pin: PhantomPinned,
+
+    /// Context initialised on construction.
+    pub context: T,
 }
 
-impl Registration {
+impl<T: Sync> Registration<T> {
     /// Creates a new [`Registration`] but does not register it yet.
     ///
     /// It is allowed to move.
-    pub fn new() -> Self {
+    pub fn new(context: T) -> Self {
         Self {
             registered: false,
             mdev: bindings::miscdevice::default(),
             _pin: PhantomPinned,
+            context,
         }
     }
 
     /// Registers a miscellaneous device.
     ///
     /// Returns a pinned heap-allocated representation of the registration.
-    pub fn new_pinned<T: FileOperations>(
+    pub fn new_pinned<F: FileOperations>(
         name: CStr<'static>,
         minor: Option<i32>,
+        state: T,
```

Comment:
Can we use the same name (either `state` or `context`) consistenty?

Reply:
Good eye, fixed.